src/App/Fossa/Analyze.hs

```diff
@@ -371,6 +381,27 @@ filterProjects rootDir = filter (isProductionPath . dropPrefix rootPath . fromAb
     dropPrefix :: String -> String -> String
     dropPrefix prefix str = fromMaybe prefix (stripPrefix prefix str)
 
+-- We want to treat IAT dependencies separately so that we can resolve them into user-defined custom dependencies.
+-- Remove any deps from the graph of type IATType.
+-- Evaluates to the list of dependencies so removed and the list of project results after removal.
+extractIATDeps :: [ProjectResult] -> ([Dependency], [ProjectResult])
+extractIATDeps results = do
+  -- hack: process the results twice to get the two pieces of infomation desired.
+  -- I'm happy to do this a better way but I'm not sure how to combine these into one action.
+  let iatDepGraphs = Graphing.filter isIATDep <$> (projectResultGraph <$> results)
+  let iatOnlyDeps = concat $ Graphing.toList <$> iatDepGraphs
+  (iatOnlyDeps, filterIATDeps <$> results)
```

Comment:
I'd love anyone's assistance on this- I'm sure there's a cleaner way to do this but I wasn't sure how to do it non-imperatively.

Reply:
I think this highlights a couple of technical design issues:

- strategies typically don't have access to `ScanDestination`, which is required for the API requests. This is already kind of solved for `VSI`, because we already pass in the `ScanDestination`. I think the `resolveAssertedDeps` could actually happen within the `VSI` strategy
- strategies don't have a way to emit `AdditionalDepData`.. which is a big contributor for why `analyzeFossaDepsFile` isn't a strategy anymore

There are two possible resolutions here:

1. Extract out the VSI strategy and run it separately as part of `analyze`, similar to `analyzeFossaDepsFile`
2. Add a field to `DependencyResults` that would allow us to emit `AdditionalDependencyData`

The first option is somewhat painful, and I'm not sure how to feel about the second option

cc @scruffystuffs

Reply:
I really don't like the second option, but I think it's because our model of what we find to what source units we output is wrong.  The entire `fossa-deps` file is wrapped into a single source unit, which was impossible in the current strategy model.

I'm less interested in how we convert to source units (and therefore, how we emit "additional dep data"), and more interested in "How do we abstractly model *every single thing we found*, during all kinds of analyses?"  I don't know if we can feasibly change this for this PR, but I think it shows we're still a little bit off the mark.

Reply:
> Extract out the VSI strategy and run it separately as part of `analyze`, similar to `analyzeFossaDepsFile`

I would prefer to not do this as part of this PR, but this might be the best path forward for future maintainability- I'm envisioning having the VSI strategy output its own `SourceUnit` directly instead of a `Graphing Dependency`. Is that what you were thinking as well?

In particular I have another somewhat substantial change coming down the pipe next, I could do this there.

Reply:
> I'm envisioning having the VSI strategy output its own SourceUnit directly instead of a Graphing Dependency. Is that what you were thinking as well?

Yes, exactly

Reply:
@cnr I ended up refactoring the VSI logic out as part of this PR after all. I didn't see a real reason to wait. Would appreciate if you want to take a look!